packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -1866,7 +1866,17 @@ export default (superClass: Class<Parser>): Class<Parser> =>
         return this.tsParseImportEqualsDeclaration(node);
       }
 
-      if (this.eatContextual("type")) {
+      const ahead = this.lookahead();
```

Comment:
We should probably only do the lookahead if `this.match(tt.name)`.

Does it make sense to combine this with the above block?

```
if (this.match(tt.name)) {
  const ahead = this.lookahead();

  if (ahead.type === tt.eq) {
    return this.tsParseImportEqualsDeclaration(node);
  }

  if (
    this.state.value === "type" && 
    // import type, { a } from "b";
    ahead.type !== tt.comma &&
    // import type from "a";
    !(ahead.type === tt.name && ahead.value === "from")
  ) {
    node.importKind = "type";
  }
} else {
  node.importKind = "value";
}
```